ulauncher/utils/fuzzy_search.py

```diff
@@ -48,19 +48,24 @@ def get_matching_indexes(query, text):
     return sorted(positions)
 
 
-def get_score(query, text):
+def get_score(query="", text=""):
     """
     Uses Levenshtein's algorithm + some improvements to the score
     :returns: number between 0 and 100
     """
-    if not query or not text:
-        return 0
-
     query = query.lower()
     text = text.lower()
-    score = ratio(query, text) * 100
-
-    # increase score if a word from text starts with a query
+    query_len = len(query)
+    if not query or not text:
+        return 0
+    if text.startswith(query):
+        return 100
+    # Wrap and counter-weight distance so that short queries can match better with long texts.
+    # With regular distance "Fir" and "Firefox" is only 3/7 similar
+    # We want them to be counted as near 100% matches (if you have an actual app named "Fir"
+    # it's better that is listed first).
```

Comment:
Doesn't the `text.startswith(query)` check above contradict this comment and the provided example? It mentions how `fir` will match to `firefox` and `fir`, and that `fir` should be scored higher. However, in practice with this `get_score()` implementation it will return 100 for both queries since it is a prefix for both. tbh I think we have to be very clever to get those two cases sorted right while only looking at each match independently. 

Perhaps some logarithmic score increase for the number of exact characters matched in the same order from the beginning of the word which quickly rises to perfect score but perhaps never reaching it. Though maybe Levenshtein's already does something similar? It's been a while since I've looked at it in detail.

Regardless this falls squarely in the "requires more tuning" that is likely an ongoing effort, and in this case at least the 2 exact prefix matches show up near the top, even if the exact relative order between them is indeterminate.

I think the algorithm seems good to go with, but I did want to point out that the comment is slightly misleading.

Reply:
Good point. You are right. I should change the comment or code. Not sure which yet

Reply:
I should probably remove text.startswith(query) actually. It's already covered below.

Reply:
Pushed as 9db6eea. I should have been more patient to merge this PR haha :D

I removed text.startswith(query), rewrote the comment to be technically correct and added a text to show that the 95% weighting does work, just as long as the query doesn't match the start of a word.